src/blueapi/service/main.py

```diff
@@ -77,10 +84,38 @@ def submit_task(
     """Submit a task to the worker."""
     task_id: str = handler.worker.submit_task(task)
     response.headers["Location"] = f"{request.url}/{task_id}"
-    handler.worker.begin_task(task_id)
     return TaskResponse(task_id=task_id)
 
 
+@app.put("/worker/task", response_model=WorkerTask)
+def update_task(
+    task: WorkerTask,
+    handler: Handler = Depends(get_handler),
+) -> WorkerTask:
+    if task.task_id is not None:
```

Comment:
I think this should also check here that there isn't already a task running on the worker and return an error status (probably 409 Conflict) with an appropriate message if there is.